docs/topics/clusterdefinitions.md

```diff
@@ -248,6 +248,7 @@ The reason for the unsightly base64-encoded input type is to optimize delivery p
 | kube-apiserver | true | 1 per master node | The Kubernetes API server validates and configures data for the api objects which include pods, services, replicationcontrollers, and others. The API Server services REST operations and provides the frontend to the cluster's shared state through which all other components interact. Official docs [here](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/). |
 | kube-scheduler | true | 1 per master node | The Kubernetes scheduler is a policy-rich, topology-aware, workload-specific function that significantly impacts availability, performance, and capacity. The scheduler needs to take into account individual and collective resource requirements, quality of service requirements, hardware/software/policy constraints, affinity and anti-affinity specifications, data locality, inter-workload interference, deadlines, and so on. Official docs [here](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/). |
 | kube-addon-manager | true | 1 per master node | Addon manager provides a standard way to deliver additional Kubernetes componentry. The addons supported by AKS Engine (documented [above](#addons)) are installed into the cluster using Addon manager. Official docs [here](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/addon-manager). |
+| cluster-init | false | n/a | This is a "backdoor" interface for delivering manual Kubernetes resource configuration that is appropriate for tight coupling to the ARM deployment specification. For example, if you are using aks-engine in an automated pipeline and you can scale up a cluster more quickly by loading a Kubernetes specification immediately after bootstrapping the control plane, you may use this interface by base64 encoding a yaml representation of that spec. The base64-encoded data string will be decoded at cluster creation time into a yaml file on the first master VM at the `/opt/azure/containers/cluster-init.yaml` filepath, and then loaded into the cluster via `kubectl apply -f`. See[here](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply) for more documentation on how to appropriate specify your source yaml for `kubectl apply` before base64 encoding it. |
```

Comment:
@Michael-Sinz How does this documentation snippet accord with your understanding of this feature?

Reply:
Looks good - I would mention that this allows us to then produce the arm template via aks-engine that when deployed, deploys a fully configured cluster *with* our full application stack.

Since the yaml is limited in size/scope, it is used to run the rest of our application deployment process as a job in the cluster.  (Maybe not what others want but it is what we do given the size and complexity of what we install, it would not be possible in a yaml that would fit into the arm template.)